### ISSUE TITLE
Change datatype of Entity.Type from Entity to object

### ISSUE THREAD

**Issue by per-samuelsson** [Starcounter/Starcounter#2521]
As we have decided to postpone #2502, as discussed in this https://github.com/Starcounter/Starcounter/issues/2502#issuecomment-71632818, I now decided that the temporary solution until we have #2502 done will be to change the datatype of ```Entity.Type``` to ```object```.

With this, we are able to drop the explicit auto-creation of the ```Entity``` during binding, and instead return one of the metadata instances. 

This also fits the changing of #2516, which I will do at the same time.

This relates to, and affects, both #2510 and #2480.

**Comment by per-samuelsson** [comment]
While working with this, and #2516 in parallel, it stroke me that it's a lot nicer to return IObjectView, and doable too; anything that can be returned - Entity-based subclasses, custom type classes defined with ```[Database]``` and all the metadata classes - should always be (and is) castable to that interface. So this is what I'm changing to.

**Comment by per-samuelsson** [comment]
> With this, we are able to drop the explicit auto-creation of the Entity during binding, and instead return one of the metadata instances.

Decided to factor out the above to a separate issue: #2526.

**Comment by per-samuelsson** [comment]
This issue is completed, minus factored out #2526, pushed and merged in https://github.com/Starcounter/Starcounter/commit/c07aa30064a962098a113b91282a969ae5692263. The set of commits pushed also extend the dynamic types regression test quite a bit.
